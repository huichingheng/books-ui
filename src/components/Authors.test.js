import React from "react";
import Authors from "./Authors";
import ShallowRender from "react-test-renderer/shallow";

test("renders Author component", () => {
  const renderer = new ShallowRender();
  renderer.render(<Authors />);
  const output = renderer.getRenderOutput();

  expect(output).toMatchSnapshot();
});

test("renders name and authorlist after the component mount",async ()=>{
  fetch.mockResponseOnce(
    JSON.stringify([
      {
        name: "Harry Potter author", 
        _id: "654723849s"
      }
    ])
  )

  const renderer = new ShallowRender();
  renderer.render(<Authors />)

  const instance = renderer.getMountedInstance()
  await instance.componentDidMount()

  const output = renderer.getRenderOutput();
  expect(output).toMatchSnapshot();
})

