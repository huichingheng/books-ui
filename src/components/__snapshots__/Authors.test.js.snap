// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Author component 1`] = `
<div>
  <h2>
    Authors
  </h2>
</div>
`;

exports[`renders name and authorlist after the component mount 1`] = `
<div>
  <h2>
    Authors
  </h2>
  <li>
    Harry Potter author
  </li>
</div>
`;
